brian2genn/templates/model.cpp

```diff
@@ -186,15 +182,19 @@ void modelDefinition(NNmodel &model)
   //  for (int i=0; i<{{openmp_pragma('get_num_threads')}}; i++)
   //    brian::_mersenne_twister_states.push_back(new rk_state());
 
-  {% for max_row_length in max_row_length_code_1 %}
+  {% for max_row_length in max_row_length_run_array %}
   {{max_row_length}}
   {% endfor %}
 
-  {% for max_row_length in max_row_length_code_2 %}
+  {% for max_row_length in max_row_length_run_generator %}
   {{max_row_length}}
   {% endfor %}
```

Comment:
Why is it necessary that all the synapses are first created via the "array" syntax and then via the "generator" syntax? Couldn't you stick both `run_...` calls into the same list and execute them all in the order the user specified? Actually if you want to use the exact same random numbers as Brian later, you'll *have* to use this order.

Reply:
Good point! While I don't think there is a problem with random numbers as the arrays don't use any, the separation is a hangover of my previous incomplete understanding of how it works (I *thought* previously that array based stuff would be combined in python and straight read into the connectivity dynamic arrays in C++ (which is not the case)) - will simplify/correct.